gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-client/src/test/java/org/apache/gobblin/service/FlowConfigV2Test.java

```diff
@@ -301,6 +302,42 @@ public void testLocalGroupOwnershipUpdates() throws Exception {
     _client.deleteFlowConfig(new FlowId().setFlowGroup(TEST_GROUP_NAME).setFlowName(TEST_FLOW_NAME_7));
   }
 
+
+  @Test (expectedExceptions = RestLiResponseException.class)
+  public void testGroupUpdateRejected() throws Exception {
+   ServiceRequester testRequester = new ServiceRequester("testName", "USER_PRINCIPAL", "testFrom");
+   _requesterService.setRequester(testRequester);
+   Map<String, String> flowProperties = Maps.newHashMap();
+
+   FlowConfig flowConfig = new FlowConfig().setId(new FlowId().setFlowGroup(TEST_GROUP_NAME).setFlowName(TEST_FLOW_NAME_8))
+       .setTemplateUris(TEST_TEMPLATE_URI)
+       .setProperties(new StringMap(flowProperties))
+       .setOwningGroup("testGroup");
+
+   _client.createFlowConfig(flowConfig);
+
+   flowConfig.setOwningGroup("testGroup2");
+   _client.updateFlowConfig(flowConfig);
```

Comment:
Why would we get exception here? From what I see, user creates a flow and then sets the owning group for the flow that he owns.

Reply:
The point is to check that a user cannot update a flow's owningGroup to a group that that user is not a part of (the test user is not part of the group `testGroup2`). Previously we only checked this condition on flow creation, but you could still update it to a group you are not part of.

Reply:
We should avoid using testGroup2 as a failing group. There's another test that writes testGroup2 to the file where the group ownership service reads from, to test if the groupownership service will update automatically on reads, and this could lead to flakiness

Reply:
Actually it looks like that test also deletes the original group (testGroup1), which was causing some confusing behaviour, and I think could affect the already existing tests as well depending on what order they run in. I made that one depend on all the ones that rely on that group being there.